Xamarin.Forms.Platform.Android/FastRenderers/AutomationPropertiesProvider.cs

```diff
@@ -142,18 +132,27 @@ bool SetHint(string hint = GetFromElement)
 				_defaultHint = textView.Hint;
 			}
 
-			string value = hint;
-			if (value == GetFromElement)
-			{
-				value = string.Join(". ", (string)Element.GetValue(AutomationProperties.NameProperty),
-					(string)Element.GetValue(AutomationProperties.HelpTextProperty));
-			}
+			string value = ConcatenateNameAndHelpText(Element);
 
 			textView.Hint = !string.IsNullOrWhiteSpace(value) ? value : _defaultHint;
 
 			return true;
 		}
 
+		static string ConcatenateNameAndHelpText(Element Element)
+		{
+			string separator;
+			var name = (string)Element.GetValue(AutomationProperties.NameProperty);
+			var helpText = (string)Element.GetValue(AutomationProperties.HelpTextProperty);
+
+			if (string.IsNullOrWhiteSpace(name) || string.IsNullOrWhiteSpace(helpText))
+				separator = "";
+			else
+				separator = ". ";
+
+			return $"{name}{separator}{helpText}";
```

Comment:
1/ Are you sure you mean `IsNullOrWhiteSpace` and not `IsNullOrEmpty` ?

2/ Should be:
```
if (string.IsNullOrWhiteSpace(name)
    return helpText;
if (string.IsNullOrWhiteSpace(helpText)
    return name;

return $"{name}. {helpText}"
```
so you don't concatenate empty strings (this 2/ is only valid if you don't care about whitespaces